### PR TITLE
build: lock closed and inactive issues via GH Action

### PR DESCRIPTION
**Pull request by NoNameProvided**
This PR adds a GH action to lock closed and inactive issues.

### CODE REVIEW

File: .github/workflows/lock-closed-issues-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: 'Lock threads'
```
Comment (saulotoledo) [inline .github/workflows/lock-closed-issues-workflow.yml:1]: I suggest to rename it to "Lock inactive threads". This only closes inactive threads which were already closed, right?

File: .github/workflows/lock-closed-issues-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          issue-lock-inactive-days: 30
+          pr-lock-inactive-days: 30
+          issue-lock-comment: >
+            This issue has been automatically locked since there
+            has not been any recent activity after it was closed.
+            Please open a new issue for related bugs.
+          pr-lock-comment: >
+            This pull request has been automatically locked since there
+            has not been any recent activity after it was closed.
+            Please open a new issue for related bugs.
```
Comment (saulotoledo) [inline .github/workflows/lock-closed-issues-workflow.yml:22]: I would add the EOF newline. Prettier is checking that on ts files, but there are still non-ts files without it in the repos.

File: .github/workflows/lock-closed-issues-workflow.yml (hunk 1 above)
Comment (NoNameProvided) [reply .github/workflows/lock-closed-issues-workflow.yml]: Fixed.

File: .github/workflows/lock-closed-issues-workflow.yml (hunk 2 above)
Comment (NoNameProvided) [reply .github/workflows/lock-closed-issues-workflow.yml]: Fixed.

**Pull request merged**